pkg/server/plugin/upstreamca/awssecret/awslib.go

```diff
@@ -0,0 +1,48 @@
+package awssecret
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
+)
+
+type awsClient interface {
```

Comment:
this is unused. consider replacing with a `secretsManagerClient` interface that has the single method needed for this plugin (i.e. `GetSecretValueWithContext`). If you do that then your fake implementation doesn't need to embed `secretsmanageriface.SecretsManagerAPI` and it also becomes really clear exactly what our dependency on AWS is.